packages/kit/src/core/adapt/prerender.js

```diff
@@ -75,14 +75,31 @@ export async function prerender({ cwd, out, log, config, build_data, fallback, a
 		read: (file) => readFileSync(join(config.kit.files.assets, file))
 	});
 
-	/** @type {(status: number, path: string, parent: string | null, verb: string) => void} */
-	const error = config.kit.prerender.force
-		? (status, path, parent, verb) => {
-				log.error(`${status} ${path}${parent ? ` (${verb} from ${parent})` : ''}`);
-		  }
-		: (status, path, parent, verb) => {
-				throw new Error(`${status} ${path}${parent ? ` (${verb} from ${parent})` : ''}`);
-		  };
+	/** @typedef {((status: number, path: string, parent: string | null, verb: string) => void | never)} InternalErrorHandler */
+	/** @type {InternalErrorHandler} */
+	const error = (function () {
```

Comment:
I am happy to rewrite this as something other than an IIFE, but it seemed a decent way to colocate everything.

Reply:
could we just make it a function that lives outside the `prerender` function?

Reply:
Happy to.